src/Facade/Services/StoresFunctionCodeService.cs

```diff
@@ -38,12 +41,23 @@ protected override void DeleteOrObsoleteResource(StoresFunction entity, IEnumera
             throw new NotImplementedException();
         }
 
-        protected override Expression<Func<StoresFunction, bool>> FilterExpression(StoresFunctionResource searchResource)
+        protected override Expression<Func<StoresFunction, bool>> FilterExpression(StoresFunctionRequestResource searchResource,
+            IEnumerable<string> privileges = null)
         {
+            if (searchResource.OnlyAllowed == true)
+            {
+                if (privileges.Any())
+                {
+                    var allowedFunctions = privileges.Where(s => s.StartsWith("stores.requisitions.")) 
+                        .Select(s => s.Substring("stores.requisitions.".Length)) 
+                        .ToList();
```

Comment:
I would have the actual filtering logic in a domain service (or at least thats the route I advised oliver to go down)

Reply:
Should clarify: 'm not saying you should change this to use a domain service since you're using the common facade filterby and just returning an expression here - Just noting that I had advised Oliver to go down a different path